### ISSUE TITLE
Lazyredraw doesn't correctly refresh ex-line after mappings

### ISSUE THREAD

**Issue by SuperCuber** [neovim/neovim#12757]
- `nvim --version`: NVIM v0.4.3
- `vim -u DEFAULTS`behaves differently? Behavior only happens with `set lazyredraw`
- Operating system/version: Ubuntu 18.04 Bionic
- Terminal name/version: xfce4-terminal
- `$TERM`: xterm-256color

## Steps to reproduce
Vimrc contents:
```vim
set lazyredraw
tnoremap <C-d> exit<CR><C-\><C-n><C-w>c
nnoremap <silent> <C-f> :test
```
#### Actual
`nvim -u my_vimrc`
Open a `:split` and `:terminal` inside it.
Enter insert(terminal) mode.
Use the mapping `<C-d>` to close the window.
`-- TERMINAL --` text at ex-line stays.

#### Expected
`-- TERMINAL --` should get cleared to represent being in normal mode.

#### Actual
`nvim -u my_vimrc`
Use the mapping `<C-f>`.
Nothing happens visually.
Press any letter.
Ex-line refreshes to reveal incomplete ex command.

#### Expected
Incomplete command should be displayed immediately after pressing `<C-f` without needing to press another key.

**Comment by justinmk** [comment]
That's life with lazyredraw

**Comment by bfredl** [comment]
Just as a note, I want to fix the "ordinary" redraw in 0.6 to be so fast that 'set lazyredraw' can be a no-op
